### PR TITLE
Don't start kernel when trying to switch kernels

### PR DESCRIPTION
**Pull request by DonJayamanne**
For #6083

### CODE REVIEW

File: src/client/datascience/jupyter/kernels/kernelDependencyService.ts [hunk 1]
```diff
@@ -131,10 +138,13 @@ export class KernelDependencyService implements IKernelDependencyService {
         });
         const installPrompt = isModulePresent ? Common.reInstall() : Common.install();
         const selectKernel = DataScience.selectKernel();
+        // Due to a bug in our code, if we don't have a resource, don't display the option to change kernels.
+        // https://github.com/microsoft/vscode-jupyter/issues/6135
+        const options = resource ? [installPrompt] : [installPrompt, selectKernel];
```
Comment (DonJayamanne) [inline src/client/datascience/jupyter/kernels/kernelDependencyService.ts:143]: I'm removing the ability to switch kernels when using `Jupyter` (non-RAW), fixing this requires a big refactor or we can just hack/patch the code, either way, I'm not keen on doing that at this stage.
TLDR - we're not passing the `resource`, how/why can be dealt with #6135

File: src/client/datascience/commands/notebookCommands.ts [hunk 2]
```diff
@@ -95,12 +97,15 @@ export class NotebookCommands implements IDisposable {
...
         }
         if (options.identity) {
+            const isLocal = isLocalLaunch(this.configService);
             // Make sure we have a connection or we can't get remote kernels.
-            const connection = await this.notebookProvider.connect({
-                getOnly: false,
-                disableUI: false,
-                resource: options.resource
-            });
+            const connection = isLocal
```
Comment (DonJayamanne) [inline src/client/datascience/commands/notebookCommands.ts:102]: When swtching kernels, we do'nt need to start or connect the notebook, thats unnecessary. 

Basically its impossible to switch kernels if you have an interpreter without ipykernel or the like. This could explain why users aren unable to start notebooks as well, they cannot change the kernel.

File: src/client/datascience/jupyter/kernels/kernelDependencyService.ts [hunk 3]
```diff
@@ -89,11 +89,17 @@ export class KernelDependencyService implements IKernelDependencyService {
...
-            const cmd =
-                getResourceType(resource) === 'notebook' && this.useNativeNb
-                    ? 'notebook.selectKernel'
-                    : Commands.SwitchJupyterKernel;
-            this.commandManager.executeCommand(cmd).then(noop, noop);
+            if (getResourceType(resource) === 'notebook' && this.useNativeNb) {
+                this.commandManager.executeCommand('notebook.selectKernel').then(noop, noop);
+            } else {
+                this.commandManager
+                    .executeCommand(Commands.SwitchJupyterKernel, {
```
Comment (DonJayamanne) [inline src/client/datascience/jupyter/kernels/kernelDependencyService.ts:96]: We 'weren't passing the right argumetns, hence kernel switching would never happen, ever.

**Review by rchiodo — APPROVED** [review]

**Review by IanMatthewHuff — APPROVED** [review]

**Review by joyceerhl — APPROVED** [review]

File: src/client/datascience/jupyter/kernels/kernelDependencyService.ts [hunk 4]
```diff
@@ -131,10 +138,19 @@ export class KernelDependencyService implements IKernelDependencyService {
...
         const selectKernel = DataScience.selectKernel();
+        // Due to a bug in our code, if we don't have a resource, don't display the option to change kernels.
+        // https://github.com/microsoft/vscode-jupyter/issues/6135
+        const options = resource ? [installPrompt, selectKernel] : [installPrompt];
+        // In the case of interactive window, due to the current code flow we get this code executed twice,
+        // hence we get two messages about ipykernel not being installed.
+        // THat's a very poor ux, one could end up with two modal dialog boxes (one after the other for interactive).
+        // hence disabling modal dialog for interactive window for now.
+        // Again to be resolved in https://github.com/microsoft/vscode-jupyter/issues/6135
+        const modal = getResourceType(resource) === 'notebook';
```
[the remark this answers is not in the document]
Comment (DonJayamanne) [reply src/client/datascience/jupyter/kernels/kernelDependencyService.ts:149]: Yes.
